# Hicks Lab Onboarding Information

## Mission Statement

Our core purpose is to develop statistical methodologies, scalable computational tools and algorithms, and open-source software that for the analysis of genomics data. Specifically, we aim to addresses statistical challenges in epigenomics, functional genomics and single-cell genomics such as the pre-processing, normalization, analysis of raw, noisy high-throughput data leading to an improved quantification and understanding of biological variability.

To accomplish this, we will write good code, perform solid and reproducible analyses, and disseminate our results widely through approachable publications and software.
We recognize that trust, both in the process and in our results, is of primary importance to the biologists that use our methods and software.
Therefore, we strive to make our source code as open and accessible as possible.
When we submit papers, we expect that the analytical code behind those papers will be something that we can be proud of.
To these ends, we will provide reviewers and the scientific community with all source code required to generate figures in the paper that result from computational analyses.

## Expectations

**Your role:** We expect that you will take primary responsibility for the success of your research project and career development.
As a member of the lab, you are expected to participate fully in the team.
When disagreements about methodological approaches arise, you recognize that these should be resolved through a solid and reproducible analysis of available data.
In general, lab members are expected to be present from 9:30AM to 4:00PM on weekdays to facilitate discussion within the group.
If you aren’t sure — ask.

**Stephanie's role:** Stephanie’s goal is to facilitate your success as well as that of your projects.
Within a project, Stephanie will serve as a sounding board for ideas, will help you plan your project, address any questions that you have, and will help to devise experiments to test your hypotheses.
To facilitate your success, Stephanie will help you to plan your training, to devise a career plan that can take you to where you want to go, to advise you on your project-risk portfolio, and to provide guidance on other elements of career and project development as needed.

**Deadlines:** Our lab has worked hard to develop a reputation for high-quality science that is well presented.
We all benefit from this reputation, but we must also work to maintain it.
In order to maintain the quality of our lab's output, we've established deadlines for various outputs. The specific deadlines for various types of outputs are:

- Manuscripts must be shared two weeks before any deadlines.
- Posters must be shared one week before the deadline for printing.
- Scientific talks based on a submitted abstract must be shared one week before the presentation.
- Meeting abstracts must be shared one week before the deadline for submission.

Lab members are given a two-day period to provide feedback on the document.
We expect that authors will then revise the document to incorporate feedback provided within the initial two-day period.
Authors are encouraged but not required to address feedback received after the initial two days, as it may not always be practicable.

This does not eliminate the need for all coauthors to approve a document.
Coauthors are not required to provide their feedback within the two-day window.
Coauthors can hold submission of a document until they approve; however, according to ICMJE guidelines extreme holds may result in a change from authorship to acknowledgement.

In the case that all feedback received within the two-day period has been addressed and all coauthors approve, the submission can proceed.

Failure to abide by these guidelines will result in missing whatever the opportunity in question is.

**Code of Conduct:** All members of the lab, along with visitors, are expected to agree with this code of conduct.
We will enforce this code.
We expect cooperation from all members to help ensuring a safe environment for everybody.
The lab is dedicated to providing a harassment-free experience for everyone, regardless of gender, gender identity and expression, age, sexual orientation, disability, physical appearance, body size, race, or religion (or lack thereof).
We do not tolerate harassment of lab members in any form.
Sexual language and imagery is generally not appropriate for any lab venue, including lab meetings, presentations, or discussions.
However, do note that we work on biological matters so work-related discussions of e.g. animal reproduction are appropriate
Harassment includes offensive verbal comments related to gender, gender identity and expression, age, sexual orientation, disability, physical appearance, body size, race, religion, sexual images in public spaces, deliberate intimidation, stalking, following, harassing photography or recording, sustained disruption of talks or other events, inappropriate physical contact, and unwelcome sexual attention
Members asked to stop any harassing behavior are expected to comply immediately.

If you are being harassed, notice that someone else is being harassed, or have any other concerns, please contact Stephanie Hicks immediately
If Stephanie is the cause of your concern, Dr. Margaret Taub (<mtaub@jhsph.edu>) is a good informal point of contact; she does not work for Stephanie and is the ombudsman in the department.
The code of conduct section is licensed under a Creative Commons Attribution 3.0 Unported License.

We expect members to follow these guidelines at any lab-related event.

**Ethics:** We expect lab members to be honest in scientific communications both within and outside the lab
We expect that lab members will design experiments in a manner that minimizes both bias and self deception
We expect that lab members will keep agreements, be careful, and share their code and results openly with the scientific community
We expect that credit will be given where credit is due, including in scientific writing
Plagiarism is not tolerated
While a full enumeration of ethical considerations is outside of the scope of this document, the [Office of Integrity at JHSPH](https://www.jhsph.edu/offices-and-services/office-of-academic-affairs/academic-integrity/index.html) provides a handbook that we recommend. In addition, please don't hesitate to raise any questions or concerns that you have at any point with Stephanie.

## Communication

### General

**Slack:** We use slack for rapid communication within the lab.
If you'd send an e-mail to someone within the lab, try a slack message instead.
This helps to keep communications in one place, and Stephanie commits to respond to slacks (not necessarily immediately, but the same guarantee is not made for e-mail).

**Social Media:** Lab members are encouraged to communicate through public social media, and if you choose to do so then you are expected to follow our code of conduct.

**Projects:** By the nature of our research, lab members will often have the opportunity to participate projects managed via private or publicly accessible source code repositories.
In these cases, lab members are expected to: follow the code of conduct; and to communicate via the project-specific medium (e.g. if Rene reported an issue on a project on GitHub, it would not be appropriate for Stephanie to reply "I'll drop by your desk and show you how to solve that.").

**IP/Openness:** This is handled in accordance with the instructions from our research sponsors and university guidance.
Lab members must follow the [JHSPH IP Policy](https://ventures.jhu.edu/the-johns-hopkins-ip-policy/).
These often allow, encourage, or require openness.
If you have concerns at any point, set up a meeting with Stephanie to discuss these concerns.

**Accounts:** Lab members are expected to have accounts for the following and be members of the specified (organizations) if applicable:

+ GitHub 
+ Slack (JHU Biostatistics, JHU Genomics)

### Meetings

**Lab Meeting:** Lab meetings are held weekly at a location at JHSPH.
We currently have a joint lab meeting between two faculty: Stephanie 
Hicks and Kasper Hansen. Students and postdocs rotate presenting 
the work in progress. For example, it could be presenting new results 
from analyses or simulation studies. It could also be figures you are 
creating for a paper and would like feedback. This is an opportunity 
to share anything that you wish to talk about with the group. This 
could be a confounding result, an interesting result, an analysis that
isn’t working, a demo of a cool new technology etc. This is your 
chance to have the group focus on and help you solve a challenge 
that you’re facing or to share something interesting that you've 
discovered with the group. 

**Individual Meetings:** We schedule weekly individual meetings.
Once you join the lab, contact Stephanie to set up a time.
These are set up for a term to accommodate class schedules. 
We don’t reschedule these meetings by default if one of the parties (Stephanie or you) are out of town, so if you do want to meet in a week but travel conflicts, contact Stephanie to reschedule.
The goal of the weekly meeting is to:

1.  Discuss challenges.
2.  Plan strategy (project related, personal career, etc).

## Source Code, Data, and Reproducibility

**Pride:** We expect lab members to sign their code.
To quote from _The Pragmatic Programmer_:

> Craftsmen of an earlier age were proud to sign their work.
You should be, too…
People should see your name on a piece of code and expect it to be solid, well written, tested, and documented.

While some code will be proof-of-concept code, it should be of a form that inspires confidence.

**Language:** We write code for our analyses in R or Python, which allows everyone in the lab to know two languages and understand analytical code.
Code for visualization can be R, Python or javascript.

**Version Control Services:** Our primary version control service is GitHub.
We expect that lab members will maintain their code in repositories under their own accounts.

**Projects that didn't work:** We expect that repositories will contain failures (e.g. proof-of-concepts that didn't work).
This is ideal.
Being able to find them will make sure we don't make the same failure twice.

**Data Management:** For publicly available data, scripts used to download and process these data should be preserved, as should the versions of items used in processing (e.g. probe to gene mappings).
These items should be version controlled.
Where possible, intermediate files of reasonable size can be stored to facilitate re-use, but the process to regenerate these files from publicly available data should be preserved.
When we generate data, they should be stored in a location where they are replicated and uploaded to the relevant database as soon as possible (e.g. GEO for gene expression, SRA for sequencing).

**Reproducibility:** We expect all lab members to maintain code that performs reproducible analyses.
This can be in the form of makefiles, shell scripts, or other automation approaches that allow analyses to be automatically performed.
We expect that these scripts, including those to generate figures in papers generated as a consequence of such analyses, will be included in source control repositories and made publicly available before or concurrent with the submission of preprint (if submitted) or manuscripts.
Combined with the review guidelines, this means that all code must have been reviewed for these documents to be submitted.

## How to Modify this Document

This is a living document.
The repository is at [GitHub](https://github.com/stephaniehicks/onboarding).
To make changes, fork, edit the files you wish, and create a pull request.

## Additional Resources

+ [code review checklist](extras/code_review_checklist.md)
+ [deployment checklist](extras/deployment_checklist.md)
+ [infrastructure](extras/infrastructure.md)
+ [newcomers info](extras/newcomers_info.md)
+ [conferences](extra/conferences.md)
